devtools/cli/src/main/java/io/quarkus/cli/Create.java

```diff
@@ -0,0 +1,131 @@
+        boolean maven = false;
+
+        @CommandLine.Option(names = { "--gradle" }, order = 6, description = "Create a Gradle project.")
+        boolean gradle = false;
+
+        @CommandLine.Option(names = {
+                "--grade-kotlin-dsl" }, order = 7, description = "Create a Gradle Kotlin DSL project.")
+        boolean gradleKotlinDsl = false;
+    }
+
+    @CommandLine.ArgGroup()
+    TargetLanguage language = new TargetLanguage();
+
+    static class TargetLanguage {
+        @CommandLine.Option(names = {
+                "--java" }, order = 8, description = "Generate Java examples. (default)")
+        boolean java = false;
+
+        @CommandLine.Option(names = {
+                "--kotlin" }, order = 9, description = "Generate Kotlin examples.")
+        boolean kotlin = false;
+
+        @CommandLine.Option(names = {
+                "--scala" }, order = 10, description = "Generate Scala examples.")
+        boolean scala = false;
+    }
+
+    @CommandLine.Parameters(arity = "0..1", paramLabel = "EXTENSION", description = "extension to add to project")
+    Set<String> extensions;
+
+    @CommandLine.ParentCommand
+    QuarkusCli parent;
+
+    @CommandLine.Spec
+    CommandLine.Model.CommandSpec spec;
+
+    @Override
+    public Integer call() throws Exception {
+        try {
+            File projectDirectory = new File(System.getProperty("user.dir"));
+
+            File projectRoot = new File(projectDirectory, artifactId);
+            if (projectRoot.exists()) {
+                spec.commandLine().getErr().println("Unable to create the project, " +
+                        "the directory " + projectRoot.getAbsolutePath() + " already exists");
+                return CommandLine.ExitCode.SOFTWARE;
+            }
+
+            SourceType sourceType = SourceType.JAVA;
+            if (targetBuildTool.gradleKotlinDsl)
+                sourceType = SourceType.KOTLIN;
+            else if (language.scala)
+                sourceType = SourceType.SCALA;
+            else if (language.kotlin)
+                sourceType = SourceType.KOTLIN;
+            else if (extensions != null && !extensions.isEmpty())
+                sourceType = CreateProject.determineSourceType(extensions);
+
+            BuildTool buildTool = BuildTool.MAVEN;
+            if (targetBuildTool.gradle)
+                buildTool = BuildTool.GRADLE;
+            else if (targetBuildTool.gradleKotlinDsl)
+                buildTool = BuildTool.GRADLE_KOTLIN_DSL;
+
+            boolean status = new CreateProject(projectRoot.toPath(),
+                    QuarkusPlatformConfig.getGlobalDefault().getPlatformDescriptor())
+                            .buildTool(buildTool)
+                            .groupId(groupId)
+                            .artifactId(artifactId)
+                            .version(version)
+                            .sourceType(sourceType)
```

Comment:
@patriot1burke fyi `sourceType` is not used by codestarts, you could remove it. What you will need to do though, is to add the language extension when `--kotlin` or `--scala` is selected.

Reply:
Don't know what "add the language extension" means...

Reply:
@patriot1burke:
- java -> default
- kotlin -> extension `quarkus-kotlin`
- scala -> extension `quarkus-scala`

in the end the `--kotlin` or `--scala` are just shortcuts.

Reply:
yes!